components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/client/authentication/BasicAuthClientAuthenticator.java

```diff
@@ -0,0 +1,249 @@
+import java.util.Map;
+
+/**
+ * This classs is esponsible for authenticating OAuth clients which are client id and secret to authenticate. This
+ * authenticator will handle basic client authentication where client id and secret are present either as
+ * Authroization header or in the body of the request.
+ */
+public class BasicAuthClientAuthenticator extends AbstractOAuthClientAuthenticator {
+
+    private static Log log = LogFactory.getLog(BasicAuthClientAuthenticator.class);
+
+    /**
+     * Returns the execution order of this authenticator
+     *
+     * @return Execution place within the order
+     */
+    @Override
+    public int getPriority() {
+        return 100;
+    }
+
+    /**
+     * @param request                 HttpServletRequest which is the incoming request.
+     * @param contentMap              Body parameter map of the request.
+     * @param oAuthClientAuthnContext OAuth client authentication context.
+     * @return Whether the authentication is successful or not.
+     * @throws OAuthClientAuthnException
+     */
+    @Override
+    public boolean authenticateClient(HttpServletRequest request, Map<String, List> contentMap, OAuthClientAuthnContext
+            oAuthClientAuthnContext) throws OAuthClientAuthnException {
+
+        validateAuthenticationInfo(request, contentMap);
+        // Sets client id from canAuthenticate to void reading them again.
+        if (StringUtils.isEmpty(oAuthClientAuthnContext.getClientId())) {
+            if (log.isDebugEnabled()) {
+                log.debug("Couldn't find client ID in context. Hence returning false");
+            }
+            return false;
+        } else {
+            try {
+                if (log.isDebugEnabled()) {
+                    log.debug("Authenticating client: " + oAuthClientAuthnContext.getClientId() + " with client " +
+                            "secret.");
+                }
+                return OAuth2Util.authenticateClient(oAuthClientAuthnContext.getClientId(),
+                        (String) oAuthClientAuthnContext.getParameter(OAuth.OAUTH_CLIENT_SECRET));
+            } catch (IdentityOAuthAdminException e) {
+                throw new OAuthClientAuthnException(OAuth2ErrorCodes.INVALID_CLIENT, "Error while authenticating " +
+                        "client", e);
+            } catch (InvalidOAuthClientException e) {
+                throw new OAuthClientAuthnException(OAuth2ErrorCodes.INVALID_CLIENT,
+                        "Invalid Client : " + oAuthClientAuthnContext.getClientId(), e);
+            } catch (IdentityOAuth2Exception e) {
+                throw new OAuthClientAuthnException(OAuth2ErrorCodes.INVALID_CLIENT, "Error while authenticating " +
+                        "client", e);
+            }
+        }
+    }
+
+    private void validateAuthenticationInfo(HttpServletRequest request, Map<String, List> contentMap)
+            throws OAuthClientAuthnException {
+        if (isAuthorizationHeaderExists(request)) {
```

Comment:
Aren't we follow the convention to put a line space between the method declaration and the first line of the method? This is relevant to some other places as well in this pr.